common/cache/lru.go

```diff
@@ -339,9 +331,10 @@ func (c *lru) deleteInternal(element *list.Element) {
 }
 
 // tryEvictUntilEnoughSpace try to evict entries until there is enough space for the new entry
-func (c *lru) tryEvictUntilEnoughSpace() {
+func (c *lru) tryEvictUntilEnoughSpace(newEntrySize int) {
 	element := c.byAccess.Back()
-	for c.currSize > c.maxSize && element != nil {
+	// currSize will be updated within deleteInternal
+	for c.currSize+newEntrySize > c.maxSize && element != nil {
 		entry := element.Value.(*entryImpl)
 		if entry.refCount == 0 {
 			c.deleteInternal(element)
```

Comment:
previously this logic will only evict one entry (evict by count)
now the logic is evict by size, so the logic should be changed to

```go
	for c.currSize+newEntrySize > c.maxSize && element != nil {
		entry := element.Value.(*entryImpl)
		if entry.refCount == 0 {
			elementPrev = element.Prev()
			c.deleteInternal(element)
			element = elementPrev
		} else {
			element = element.Prev()
		 }
	}
}
```